libs/components/packages/src/schematics/rules/remove-unused-dependencies.ts

```
@@ -0,0 +1,182 @@
+    const packages = Array.isArray(packageNames)
+      ? packageNames
+      : [packageNames];
+
+    const usedPackages = new Set<string>();
+
+    // Scan all projects once to find which packages are actually used
+    workspace.projects.forEach((project) => {
+      const sourceRoot = getSourceRoot(project);
+      const projectUsedPackages = findUsedPackages(tree, packages, sourceRoot);
+
+      projectUsedPackages.forEach((pkg) => usedPackages.add(pkg));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)